test/WebJobs.Script.Tests/Description/FunctionAppValidationServiceTests.cs

```diff
@@ -0,0 +1,151 @@
+
+            Assert.Null(traceMessage);
+        }
+
+        [Fact]
+        public async Task StartAsync_PlaceholderMode_DoesNotLogError()
+        {
+            _testLoggerProvider.ClearAllLogMessages();
+
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(EnvironmentSettingNames.AzureWebsitePlaceholderMode, "1");
+
+            var service = new FunctionAppValidationService(
+                _testLogger,
+                _scriptOptionsMock.Object,
+                environment);
+
+            // Act
+            await service.StartAsync(CancellationToken.None);
+
+            //Assert
+            var traces = _testLoggerProvider.GetAllLogMessages();
+            var traceMessage = traces.FirstOrDefault(val => val.EventId.Name.Equals("MissingAzureFunctionsFolder"));
+
+            Assert.Null(traceMessage);
+        }
+
+        [Fact]
+        public async Task StartAsync_NewAppWithNoPayload_DoesNotLogError()
+        {
+            _testLoggerProvider.ClearAllLogMessages();
+
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(EnvironmentSettingNames.FunctionWorkerRuntime, "dotnet-isolated");
+
+            var scriptOptionsMock = new Mock<IOptions<ScriptJobHostOptions>>();
+
+            var scriptJobHostOptions = new ScriptJobHostOptions
+            {
+                RootScriptPath = "test-root-path",
+                IsDefaultHostConfig = true
+            };
+
+            scriptOptionsMock.Setup(o => o.Value).Returns(scriptJobHostOptions);
+
+            var service = new FunctionAppValidationService(
+                _testLogger,
+                scriptOptionsMock.Object,
+                environment);
+
+            // Act
+            await service.StartAsync(CancellationToken.None);
+
+            //Assert
+            var traces = _testLoggerProvider.GetAllLogMessages();
+            var traceMessage = traces.FirstOrDefault(val => val.EventId.Name.Equals("MissingAzureFunctionsFolder"));
+
+            Assert.Null(traceMessage);
+        }
+
+        [Fact]
+        public async Task StartAsync_MissingAzureFunctionsFolder_LogsWarning()
+        {
+            _testLoggerProvider.ClearAllLogMessages();
+
+            // Arrange
+
+            Directory.CreateDirectory("test-root-path");
+            var functionMetadataList = ImmutableArray.Create(new FunctionMetadata());
+
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(EnvironmentSettingNames.FunctionWorkerRuntime, "dotnet-isolated");
+
+            var service = new FunctionAppValidationService(
+                _testLogger,
+                _scriptOptionsMock.Object,
+                environment);
+
+            // Act
+            await service.StartAsync(CancellationToken.None);
+
+            await Task.Delay(_initializationDelay);
```

Comment:
This may lead to unreliable behavior. There's a helper method used in other tests that waits for specific log messages from the host, we can use that instead.

Reply:
Makes sense. Updated.